pkg/compat/fake/client.go

```diff
@@ -0,0 +1,59 @@
+package fake
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/konveyor/mig-controller/pkg/compat"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	krand "k8s.io/apimachinery/pkg/util/rand"
+	dapi "k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+type fakeCompatClient struct {
+	k8sclient.Client
+	dapi.DiscoveryInterface
+}
+
+func (fake fakeCompatClient) RestConfig() *rest.Config {
+	return nil
+}
+func (fake fakeCompatClient) MajorVersion() int {
+	return 0
+}
+func (fake fakeCompatClient) MinorVersion() int {
+	return 0
+}
+func (fake fakeCompatClient) Get(ctx context.Context, key k8sclient.ObjectKey, obj runtime.Object) error {
+	return fake.Client.Get(ctx, key, obj)
+}
+func (fake fakeCompatClient) List(ctx context.Context, opt *k8sclient.ListOptions, obj runtime.Object) error {
+	return fake.Client.List(ctx, opt, obj)
+}
+func (fake fakeCompatClient) Create(ctx context.Context, obj runtime.Object) error {
+	switch obj.(type) {
+	case *corev1.Pod:
```

Comment:
by default, k8s fake client implementation cannot generate names on resources through `generateName` which is required for Rsync pod creation logic. This is a rudimentary way of achieving that

Reply:
+1